test/jdk/java/foreign/TestLayoutPaths.java

```diff
@@ -98,6 +99,18 @@ public void testUnknownByteStructField() {
         g.byteOffset(groupElement("foo"));
     }
 
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testTooBigGroupElementIndex() {
+        GroupLayout g = MemoryLayout.structLayout(JAVA_INT);
+        g.byteOffset(groupElement(1));
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testNegativeGroupElementIndex() {
+        GroupLayout g = MemoryLayout.structLayout(JAVA_INT);
+        g.byteOffset(groupElement(-1));
+    }
+
```

Comment:
Maybe you could add another test case that tests an index that is too large. e.g. `2` for a struct with 1 element.

Reply:
Good point - it was my intention but got distracted :-)

Reply:
actually... there is such a test:

```
@Test(expectedExceptions = IllegalArgumentException.class)
    public void testTooBigGroupElementIndex() {
        GroupLayout g = MemoryLayout.structLayout(JAVA_INT);
        g.byteOffset(groupElement(1));
    }
```

Reply:
Ah sorry! I misread that code as selecting the first element 📦